src/backend.md

```diff
@@ -18,6 +19,236 @@
 Table of contents can be generated in services like http://ecotrust-canada.github.io/markdown-toc/
 -->
 
+## Project structure
+
+Project structure should follow next principles:
+
+1. Not needed to move files without changing their purpose.
+
+1. It's easy to identify a single place where a new file should be located.
+
+1. It's easy to understand where to look files/functionality up.
+
+### Top level basic project structure
+
+- src
+  - controllers
+  - events
+  - jobs
+  - lib
+  - models
+  - sdk
+  - services
+  - tests
+
+#### controllers directory
+
+1. Controllers for resources should be placed into separated directories named as resource. For example: `user`, `role`, e.t.c.
+2. Each resource directory should have 4 files. For example if resource is `user`:
+
+| File extension            | description                        |
+| ------------------------- | ---------------------------------- |
+| `user.controller.ts`      | controller handlers implementation |
+| `user.controller.test.ts` | tests for controller handlers      |
+| `user.endpoints.ts`       | endpoints declaration              |
+| `handlers.ts`             |                                    |
+
+`handlers.ts` - special file which contains only 1 export instruction:
+
+```ts
+export * from './user.controller';
+```
+
+Controllers directory structure example:
+
+```
+- controllers
+  - user
+    - user.controller.ts
+    - user.controller.test.ts
+    - user.endpoints.ts
+    - handlers.ts
+  - endpoints.ts
+  - index.ts
+```
+
+For some resources `*.controller.test.ts` file can become huge (> 1500 rows). So if it exceeds some comfortable size for reading it may be splitted into separate files and put them into `tests` directory inside resource directory.
+
+Assume that user resource has following endpoint names: `getUsers`, `getUser`, `createUser`. So each test file should be named as `{endpoint name}.controller.test.ts`.
+
+Example:
+
+```
+- controllers
+  - user
+    - tests
+      - getUsers.controller.test.ts
+      - getUser.controller.test.ts
+      - createUser.controller.test.ts
+    - user.controller.ts
+    - user.endpoints.ts
+    - handlers.ts
+  - endpoints.ts
+  - index.ts
+```
+
+#### events directory
+
+Event handlers for resources should be placed into separated directories named as resource. For example: `user`, `role`, e.t.c.
```

Comment:
add prefixes to the files. It's needed since we handle all events from all micro services in a given service. E.g. we can handle both `crDocument` and `acDocument` updates